src/herder/HerderImpl.cpp

```diff
@@ -125,14 +125,25 @@ HerderImpl::getStateHuman() const
 void
 HerderImpl::bootstrap()
 {
-    CLOG(INFO, "Herder") << "Force joining SCP with local state";
-    assert(getSCP().isValidator());
-    assert(mApp.getConfig().FORCE_SCP);
+    if (!mBootstrapped)
+    {
+        mBootstrapped = true;
+
+        CLOG(INFO, "Herder") << "Force joining SCP with local state";
+        assert(getSCP().isValidator());
+
+        if (!mApp.getConfig().FORCE_SCP)
```

Comment:
Should we just always restore SCP state in the code path `mHerder->restoreState()` and not do it arbitrarly late: the only difference between `FORCE_SCP` and no `FORCE_SCP` will then only be that we trigger a ledger  or not.

Reply:
I did that initially, but then moved "restore state" path to later for the following reasons:
* If we move this to `mHerder->restoreState()`, we'd need to setup a timer to properly go out of sync as `restoreState` is used for normal node operation as well. For the simulation purposes, I don't think we want that, as nodes should stay "in sync" until manualclose is called (which can take arbitrarily long) in order to trigger the next ledger 
* minor: there is an existing optimization to not go in sync when we're at the genesis ledger and waiting for consensus  

I could change it as per your suggestion, but it seems like this way we get more flexibility. What do you think?